src/index.js

```diff
@@ -58,19 +58,33 @@ function setValue(meta, dataset, key, value) {
   }
 }
 
-const getScale = Chart.version
-  ? (meta, axis) => meta[axis + 'Scale']
-  : (meta, axis) => meta.controller['_' + axis + 'Scale'];
-
-const areaIsValid = (area) => area && area.right > area.left && area.bottom > area.top;
+function getStateOptions(state, meta, key, datasetIndex) {
+  let stateOptions = state.options.get(key);
+  if (!stateOptions) {
+    stateOptions = [];
+    state.options.set(key, stateOptions);
+  } else if (!meta.hidden) {
+    stateOptions = stateOptions.filter((el) => el.datasetIndex !== datasetIndex);
+    state.options.set(key, stateOptions);
+  }
+  return stateOptions;
+}
 
 export default {
   id: 'gradient',
+
+  beforeInit(chart) {
+    const state = new Object();
```

Comment:
So JAVA :)

`new Object()` is equivalent to `{}`, except its probably slower.

Reply:
> So JAVA :)

Yes... be patient but sometimes I forgot that I'm developing in JS ;)

Reply:
done